osbrain/tests/test_agent_sync_publications.py

```diff
@@ -283,3 +283,41 @@ def test_wait(nsproxy):
     assert wait_agent_attr(client, name='error_log', length=1, timeout=0.5)
     assert server.get_attr('received') == [fast, slow]
     assert 'x' + str(slow) not in client.get_attr('received')
+
+
+def test_various_requests(nsproxy):
+    """
+    Requests arriving when server is already busy should be handled afterwards.
+
+    When the server is busy handling a request, and another client makes
+    another request, that second request should be handled after the server
+    finishes with the first request.
+    """
+    server = run_agent('server', base=ServerLate)
+    client_one = run_agent('client_one', base=Client)
+    client_two = run_agent('client_two', base=Client)
+
+    delay = 0.5
+
+    # Connect clients
+    server_addr = server.addr('publish')
+    client_one.connect(server_addr, alias='sub', handler=receive)
+    client_two.connect(server_addr, alias='sub', handler=receive)
+
+    # Each client makes a request, which should overlap in the server
+    client_one.unsafe.send('sub', delay, handler=receive)
+    time.sleep(0.1)
+    assert server.get_attr('received') == [delay]
+    client_two.unsafe.send('sub', delay, handler=receive)
+    time.sleep(0.1)
+    assert server.get_attr('received') == [delay]  # Still serving client_one
```

Comment:
Do we really need to use `.unsafe` here? I don't think so, because the request is asynchronous, but maybe I am missing something (didn't run the test myself).